Cannot compile egs_mesh on macOS with GCC 12

`egs_mesh` fails to compile on macOS (Ventura 13.1) with GCC 12 from `brew`.

Relevant section of the log:

```
g++-12 -I/Users/marc/dev/EGSnrc/HEN_HOUSE/egs++/../lib/macos -I/Users/marc/dev/EGSnrc/HEN_HOUSE/egs++/ -DOSX  -DBUILD_EGS_MESH_DLL -O2 -mtune=native  -std=c++14 -c -o /Users/marc/dev/EGSnrc/HEN_HOUSE/egs++/dso/macos/egs_mesh.o egs_mesh.cpp
In file included from egs_mesh.cpp:54:
egs_mesh.h: In member function 'void EGS_Mesh::printElement(int, std::ostream&) const':
egs_mesh.h:309:57: error: no match for 'operator[]' (operand types are 'const __gnu_cxx::__alloc_traits<std::allocator<std::array<int, 4> >, std::array<int, 4> >::value_type' {aka 'const std::array<int, 4>'} and 'int')
  309 |                  << " \t\tOn face 0: " << neighbours_[i][0] << "\n"
      |                                                         ^
egs_mesh.h:310:57: error: no match for 'operator[]' (operand types are 'const __gnu_cxx::__alloc_traits<std::allocator<std::array<int, 4> >, std::array<int, 4> >::value_type' {aka 'const std::array<int, 4>'} and 'int')
  310 |                  << " \t\tOn face 1: " << neighbours_[i][1] << "\n"
      |                                                         ^
egs_mesh.h:311:57: error: no match for 'operator[]' (operand types are 'const __gnu_cxx::__alloc_traits<std::allocator<std::array<int, 4> >, std::array<int, 4> >::value_type' {aka 'const std::array<int, 4>'} and 'int')
  311 |                  << " \t\tOn face 2: " << neighbours_[i][2] << "\n"
      |                                                         ^
egs_mesh.h:312:57: error: no match for 'operator[]' (operand types are 'const __gnu_cxx::__alloc_traits<std::allocator<std::array<int, 4> >, std::array<int, 4> >::value_type' {aka 'const std::array<int, 4>'} and 'int')
  312 |                  << " \t\tOn face 3: " << neighbours_[i][3] << "\n"
      |                                                         ^
egs_mesh.h: In member function 'EGS_Mesh::Nodes EGS_Mesh::element_nodes(int) const':
egs_mesh.h:366:35: error: no match for 'operator[]' (operand types are 'const std::array<int, 4>' and 'int')
  366 |             nodes_.at(node_indices[0]),
      |                                   ^
egs_mesh.h:367:35: error: no match for 'operator[]' (operand types are 'const std::array<int, 4>' and 'int')
  367 |             nodes_.at(node_indices[1]),
      |                                   ^
egs_mesh.h:368:35: error: no match for 'operator[]' (operand types are 'const std::array<int, 4>' and 'int')
  368 |             nodes_.at(node_indices[2]),
      |                                   ^
egs_mesh.h:369:35: error: no match for 'operator[]' (operand types are 'const std::array<int, 4>' and 'int')
  369 |             nodes_.at(node_indices[3])
      |                                   ^
make[1]: *** [/Users/marc/dev/EGSnrc/HEN_HOUSE/egs++/dso/macos/egs_mesh.o] Error 1
make: *** [egs_mesh] Error 2
```

@mxxo Any suggestions?

I may have done it wrong, but now I get something like this (just showing one such error now):

```
In file included from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/bits/unique_ptr.h:36,
                 from /opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/memory:76,
                 from egs_mesh.h:59:
/opt/homebrew/Cellar/gcc/12.2.0/include/c++/12/tuple:1595:45: note: declaration of 'struct std::array<int, 4>'
 1595 |   template<typename _Tp, size_t _Nm> struct array;
      |                                             ^~~~~
egs_mesh.h:367:23: error: invalid use of incomplete type 'const struct std::array<int, 4>'
  367 |             nodes_.at(node_indices.at(1)),
```

--- a/HEN_HOUSE/egs++/geometry/egs_mesh/egs_mesh.h
+++ b/HEN_HOUSE/egs++/geometry/egs_mesh/egs_mesh.h
@@ -59,6 +59,7 @@
 #include <memory>
 #include <sstream>
 #include <vector>
+#include <array>
 
 #include "egs_base_geometry.h"
 #include "egs_vector.h"

